WordPress/Classes/ViewRelated/Post/PostListConflictResolver.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+
+struct PostListConflictResolver {
+    /// Presents an alert with an option to discard the local or web version of a post, and opens the editor with the non-discarded version
+    static func handle(post: Post, in postListViewController: PostListViewController) {
+        let conflictResolutionAlert = presentConflictResolutionAlert(for: post) { discardWeb in
+            guard ReachabilityUtils.isInternetReachable() else {
+                let offlineMessage = NSLocalizedString("Unable to resolve version conflicts while offline. Please try again later.", comment: "Message that appears when a user tries to resolve a post version conflict while their device is offline.")
+                ReachabilityUtils.showNoInternetConnectionNotice(message: offlineMessage)
+                return
+            }
+            discardWeb ? keepLocalVersion(post: post) : keepWebVersion(post: post)
+            PostListHelper.openEditor(with: post, loadAutosaveRevision: false, in: postListViewController)
+        }
+        postListViewController.present(conflictResolutionAlert, animated: true)
+    }
+
+    /// Discard web revision, load post with local changes
+    private static func keepLocalVersion(post: Post) {
+        PostCoordinator.shared.autoSave(post)
+    }
+
+    /// Discard local changes, load latest web revision
+    private static func keepWebVersion(post: Post) {
+        post.updateRevision()
+    }
+
+    /// An alert that is presented when a post has a version conflict, and the user needs to select discarding either the local or web version
+    static func presentConflictResolutionAlert(for post: Post,
+                                                       didTapOption: @escaping (_ keepLocal: Bool) -> Void) -> UIAlertController {
+        let title = NSLocalizedString("Resolve sync conflict", comment: "Title for an alert giving the user to the option to discard the web or local version of a post.")
+
+        var localDateString = ""
+        var webDateString = ""
+
+        if let localDate = post.dateModified {
+            localDateString = PostListHelper.dateAndTime(for: localDate)
+        }
+        if let webDate = post.original?.dateModified {
+            webDateString = PostListHelper.dateAndTime(for: webDate)
+        }
+
+        let str = """
+        This post has two versions that are in conflict. Select the version you would like to discard.
```

Comment:
I'm afraid that this `String` will not be extracted by our i18n script. You can see a few examples of formatting strings, like here: https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Extensions/UIViewController+Notice.swift#L18